src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java

```diff
@@ -500,6 +502,96 @@ public void testAlColumnsLargeBatch() throws Exception {
         }
     }
 
+    @Test
+    public void testIllegalNumberOfArgNoColumnList() throws Exception {
+        String invalid = "insert into " + tableName + " values (?, ?,? ,?) ";
+
+        try (Connection connection = DriverManager.getConnection(connectionString + ";useBulkCopyForBatchInsert=true;");
+                SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) connection.prepareStatement(invalid);
+                Statement stmt = (SQLServerStatement) connection.createStatement();) {
+            Field f1 = SQLServerConnection.class.getDeclaredField("isAzureDW");
+            f1.setAccessible(true);
+            f1.set(connection, true);
+
+            pstmt.setInt(1, 1);
+            pstmt.setInt(2, 1);
+            pstmt.setInt(3, 1);
+            pstmt.setInt(4, 1);
+            pstmt.addBatch();
+
+            pstmt.executeBatch();
+            throw new Exception("shouldn't come here.");
+        } catch (BatchUpdateException e) {
+            assertEquals(e.getMessage(), "Column name or number of supplied values does not match table definition.");
+        }
+
+        invalid = "insert into " + tableName + " (c1, c2, c3) values (?, ?,? ,?) ";
+
+        try (Connection connection = DriverManager.getConnection(connectionString + ";useBulkCopyForBatchInsert=true;");
+                SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) connection.prepareStatement(invalid);
+                Statement stmt = (SQLServerStatement) connection.createStatement();) {
+            Field f1 = SQLServerConnection.class.getDeclaredField("isAzureDW");
+            f1.setAccessible(true);
+            f1.set(connection, true);
+
+            pstmt.setInt(1, 1);
+            pstmt.setInt(2, 1);
+            pstmt.setInt(3, 1);
+            pstmt.setInt(4, 1);
+            pstmt.addBatch();
+
+            pstmt.executeBatch();
+            throw new Exception("shouldn't come here.");
```

Comment:
Can we throw "Test did not throw an exception when it was Expected" instead? I think there's existing tests that do this and we should be consistent.